### PR TITLE
releases: add TiDB 5.0.0-rc release notes

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

add TiDB 5.0.0-rc release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by hunterlxt** [comment]
## TiKV
- Enable compaction guard by default, to split rocksdb SST files at TiKV region boundaries, to reduce overall compaction IO.
- Support dynamically changing auto-tuned mode of RocksDB rate limiter.
- Use grpc-rs custom checker to explicit error message for CN verification error.
- rocksdb: Fix bottommost level compression ratio lower than expected by forcing the bottommost level to use zstd compression by default.
- Fix some logs not being redact when `security.redact-info-log = true`

**Comment by crazycs520** [comment]
# TIDB

## New Features

- Support collation `utf8mb4_unicode_ci` and `utf8_unicode_ci`.
  - [project](https://github.com/pingcap/tidb/issues/17596)
  - [document](https://docs.pingcap.com/tidb/dev/character-set-and-collation#new-framework-for-collations)
- Support error/info log desensitization.
  - [project](https://github.com/pingcap/tidb/issues/18566)
  - [document](https://github.com/pingcap/tidb/blob/master/errno/logredaction.md)
- Support Invisible Indexes.
  - [project](https://github.com/pingcap/tidb/issues/9246)
  - [document](https://github.com/pingcap/tidb/pull/15366)

## Experimental Features

- Support Async commit. Async Commit is an optimization that reduces commit latency, by returning success to the user when all prewrites have succeeded.
  - [project](https://github.com/tikv/tikv/projects/34)
  - [document](https://github.com/pingcap/docs-cn/pull/5181)
- Support clustered Index. Clustered indexes provide TiDB the ability to organize tables in a way that can improve the performance of certain queries.
  - [project](github.com/pingcap/tidb/projects/45)
  - [document](https://docs.pingcap.com/tidb/dev/clustered-indexes)
- Support `LIST PARTITION` and `LIST COLUMNS PARTITION` table.
  - [project](https://github.com/pingcap/tidb/issues/20678)
  - [document](https://docs.pingcap.com/zh/tidb/dev/partitioned-table#list-%E5%88%86%E5%8C%BA) // TODO: use english doc.
- Improve the Accuracy and Robustness of Index Selection.
  - [#21817](https://github.com/pingcap/tidb/pull/21817)
  - [document](https://github.com/pingcap/docs-cn/pull/5164)

## Improvements

- improve the executor runtime information collection.
  - [issue](https://github.com/pingcap/tidb/issues/18663)
  - [document](https://docs.pingcap.com/zh/tidb/stable/sql-statement-explain-analyze#explain-analyze)
- Optimize the Performance of Bulk Deletion.
  - [issue](https://github.com/pingcap/tidb/issues/18028)

File: releases/release-5.0.0-rc.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+TiDB version: 5.0.0-rc
+
+## Compatibility Changes
+
+
+## New Features
+
+
+## Improvements
+
```
Comment (zanmato1984) [inline releases/release-5.0.0-rc.md:18]:
```suggestion

+ TiFlash
    - Added a thread pool that queues coprocessor tasks, to ease the memory pressure caused by large concurrent coprocessor handling
```

File: releases/release-5.0.0-rc.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Release date: December 31, 2020
+
+TiDB version: 5.0.0-rc
+
+## Compatibility Changes
+
+
+## New Features
+
+
```
Comment (flowbehappy) [inline releases/release-5.0.0-rc.md:16]:
```suggestion
+ TiFlash
    - Support limiting the memory usage of DeltaIndex
    - Support limiting the IO write throughput of background data management tasks, to lower the impact of foreground tasks
    
```

File: releases/release-5.0.0-rc.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        - [project](https://github.com/pingcap/tidb/issues/20678)
+        - [document](https://docs.pingcap.com/zh/tidb/dev/partitioned-table#list-%E5%88%86%E5%8C%BA) // TODO: use english doc.
+    - Improve the Accuracy and Robustness of Index Selection (Experimental)
+        - [#21817](https://github.com/pingcap/tidb/pull/21817)
+        - [document](https://github.com/pingcap/docs-cn/pull/5164)
+
++ TiKV
+
+    - Support dynamically changing auto-tuned mode of RocksDB rate limiter
+
```
Comment (HunDunDM) [inline releases/release-5.0.0-rc.md:40]:
```suggestion

+ PD

    - Support to use of the region score formula v2 to make the balance converge faster and more stable.
        - [#3269](https://github.com/tikv/pd/pull/3269)
        - [document](https://docs.pingcap.com/tidb/dev/pd-configuration-file#region-score-formula-version)
    - Support error/info log desensitization.
        - [project](https://github.com/pingcap/tidb/issues/18566)
        - [#2852](https://github.com/tikv/pd/issues/2852)
        - [document](https://docs.pingcap.com/tidb/dev/pd-configuration-file#redact-info-log)
    - Support to handled SplitRegions. [#2965](https://github.com/tikv/pd/issues/2965)
    - Supports the use of joint consensus for region scheduling. (Experimental)
        - [project](https://github.com/tikv/tikv/issues/7587)
        - [document](https://docs.pingcap.com/tidb/dev/pd-configuration-file#enable-joint-consensus-new-in-v500-rc)

```

File: releases/release-5.0.0-rc.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - Improve the executor runtime information collection
+        - [issue](https://github.com/pingcap/tidb/issues/18663)
+        - [document](https://docs.pingcap.com/zh/tidb/stable/sql-statement-explain-analyze#explain-analyze)
+    - Optimize the Performance of Bulk Deletion
+        - [issue](https://github.com/pingcap/tidb/issues/18028)
+
++ TiKV
+
+    - Enable compaction guard by default, to split rocksdb SST files at TiKV region boundaries, to reduce overall compaction I/O
+
```
Comment (HunDunDM) [inline releases/release-5.0.0-rc.md:59]:
```suggestion

+ PD

    - Enable cross table merge by default to reduce the impact of empty regions on schedule. [#3129](https://github.com/tikv/pd/pull/3129)
    - Enable placement rules by default [#3006](https://github.com/tikv/pd/pull/3006)
    - Support to assign peer roles when performing transfer-region [#3047](https://github.com/tikv/pd/pull/3047)

```

**Comment by TomShawn** [comment]
This PR is replaced by https://github.com/pingcap/docs/pull/4600.

**Pull request closed without merging**